README.md
# 0bin.cf-PasteBin-clone

Demo Page: [0bin.cf!](http://0bin.cf/)

**Author of ZeroBin:** https://github.com/sebsauvage/ZeroBin/

Modified to increase speed: 
- Original ZeroBin: http://gtmetrix.com/reports/www.zerobin.net/uusaGhh7
- Modified ZeroBin: http://gtmetrix.com/reports/0bin.cf/Vjt27nSX


Updated to Alpha 0.20 (2015-05-18) by demoz(DemogorgonZ): Improved load speed and reduced impact on server.

#Original ReadMe:

ZeroBin 0.19 Alpha

==== THIS IS ALPHA SOFTWARE - USE AT YOUR OWN RISKS ====

ZeroBin is a minimalist, opensource online pastebin where the server has zero knowledge of pasted data. Data is encrypted/decrypted in the browser using 256 bits AES.

More information on the project page: http://sebsauvage.net/wiki/doku.php?id=php:zerobin

Copyright (c) 2012 Sébastien SAUVAGE (sebsauvage.net)

This software is provided 'as-is', without any express or implied warranty. In no event will the authors be held liable for any damages arising from the use of this software.

Permission is granted to anyone to use this software for any purpose, including commercial applications, and to alter it and redistribute it freely, subject to the following restrictions:

1. The origin of this software must not be misrepresented; you must 
   not claim that you wrote the original software. If you use this 
   software in a product, an acknowledgment in the product documentation
   would be appreciated but is not required.

2. Altered source versions must be plainly marked as such, and must 
   not be misrepresented as being the original software.

3. This notice may not be removed or altered from any source distribution.


CHANGELOG.md
# ZeroBin version history #

  * **Alpha 0.8 (2012-04-11):**
    * Source code not published yet.
    * Interface completely redesigned. Icons added.
    * Now properly supports IE6/7 (ugly display, but it works. "Clone" button is disabled though.)
    * Added one level of depth for storage directories (This is better for higher load servers).
    * php version is now checked (min: 5.2.6)
    * Better checks on posted json data on server.
    * Added "1 year" expiration.
    * URLs are now converted to clickable links. This include http, https, ftp and magnet links.
    * Clickable links include ''rel="nofollow"'' to discourage SEO.
    * On my public service (http://sebsauvage.net/paste/)
      * All data will be deleted (you were warned - this is a test service)
      * Default paste expiration is now 1 month to prevent clogging-up my host.
  * **Alpha 0.9 (2012-04-11):**
    * Oh bummer... IE 8 is as shitty as IE6/7: Its does not seem to support ''white-space:pre-wrap'' correctly. I had to activate the special handling mode. I still have to test IE 9.
  * **Alpha 0.10 (2012-04-12):**
    * IE9 does not seem to correctly support ''pre-wrap'' either. Special handling mode activated for all version of IE<10. (Note: **ALL other browsers** correctly support this feature.) 
  * **Alpha 0.11 (2012-04-12):**
    * Automatically ignore parameters (such as &utm_source=...) added //after// the anchor by some stupid Web 2.0 services.
    * First public release.
  * **Alpha 0.12 (2012-04-18):**
    * **DISCUSSIONS !** Now you can enable discussions on your pastes. Of course, posted comments and nickname are also encrypted and the server cannot see them.
    * This feature implies a change in storage format. You will have to delete all previous pastes in your ZeroBin. 
    * Added [[php:vizhash_gd|Vizhash]] as avatars, so you can match posters IP addresses without revealing them. (Same image = same IP). Of course the IP address cannot be deduced from the Vizhash.
    * Remaining time before expiration is now displayed.
    * Explicit tags were added to CSS and jQuery selectors (eg. div#aaa instead of #aaa) to speed up browser. 
    * Better cleaning of the URL (to make sure the key is not broken by some stupid redirection service)
  * **Alpha 0.13 (2012-04-18):**
    * FIXED: ''imageantialias()'' call removed because it's not really usefull and can be a problem on most hosts (if GD is not compiled in php).
    * FIXED: $error not properly initialized in index.php
  * **Alpha 0.14 (2012-04-20):**
    * ADDED: GD presence is checked. 
    * CHANGED: Traffic limiter data files moved to data/ (→easier rights management)
    * ADDED: "Burn after reading" implemented. Opening the URL will display the paste and immediately destroy it on server.
  * **Alpha 0.15 (2012-04-20):**
    * FIXED: 2 minor corrections to avoid notices in php log.
    * FIXED: Sources converted to UTF-8.
  * **Alpha 0.15 (2012-04-20):**
    * FIXED: 2 minor corrections to avoid notices in php log.
    * FIXED: Sources converted to UTF-8.
  * **Alpha 0.16**:
    * FIXED minor php warnings.
    * FIXED: zerobin.js reformated and properly commented.
    * FIXED: Directory structure re-organized.
    * CHANGED: URL shortening button was removed. (It was bad for privacy.)
  * **Alpha 0.17 (2013-02-23)**:
    * ADDED: Deletion URL.
    * small refactoring.
    * improved regex checks.
    * larger server alt on installation.    
  * **Alpha 0.18 (2013-02-24)**:
    * ADDED: The resulting URL is automatically selected after pressing "Send". You just have to press CTRL+C.
    * ADDED: Automatic syntax highlighting for 53 languages using highlight.js
    * ADDED: "5 minutes" and "1 week" expirations.
    * ADDED: "Raw text" button.
    * jQuery upgraded to 1.9.1
    * sjcl upgraded to GitHub master 2013-02-23
    * base64.js upgraded to 1.7
    * FIXED: Dates in discussion are now proper local dates.
    * ADDED: Robot meta tags in HTML to prevent search engines indexing.
    * ADDED: Better json checking (including entropy).
    * ADDED: Added version to js/css assets URLs in order to prevent some abusive caches to serve an obsolete version of these files when ZeroBin is upgraded.
    * "Burn after reading" option has been moved out of Expiration combo to a separate checkbox. Reason is: You can prevent a read-once paste to be available ad vitam eternam on the net.
  * **Alpha 0.19 (2013-07-05)**:
    * Corrected XSS security flaw which affected IE<10. Other browsers were not affected.
    * Corrected spacing display in IE<10.
  * **Alpha 0.20 (2015-05-18) by demoz**: Improved load speed and reduced impact on server.


js/base64.js
!function(r){var t="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/",n=function(){for(var r=[],t="A".charCodeAt(0),n="a".charCodeAt(0),e="0".charCodeAt(0),o=0;26>o;o++)r.push(t+o);for(var o=0;26>o;o++)r.push(n+o);for(var o=0;10>o;o++)r.push(e+o);return r.push("+".charCodeAt(0)),r.push("/".charCodeAt(0)),r}(),e=function(r){for(var t={},n=0,e=r.length;e>n;n++)t[r.charAt(n)]=n;return t}(t),o=function(r){for(var t=[],n=0,e=r.length;e>n;n++)t[n]=r.charCodeAt(n);return t},a=function(r){for(var t=0;r.length%3;)r.push(0),t++;for(var e=[],o=0,a=r.length;a>o;o+=3){var u=r[o],c=r[o+1],i=r[o+2];if(u>=256||c>=256||i>=256)throw"unsupported character found";var T=u<<16|c<<8|i;e.push(n[T>>>18],n[T>>>12&63],n[T>>>6&63],n[63&T])}for(;t--;)e[e.length-t-1]="=".charCodeAt(0);return F(e)},u=function(r){r=r.replace(/[^A-Za-z0-9+\/]+/g,"");for(var t=[],n=r.length%4,o=0,a=r.length;a>o;o+=4){var u=(e[r.charAt(o)]||0)<<18|(e[r.charAt(o+1)]||0)<<12|(e[r.charAt(o+2)]||0)<<6|(e[r.charAt(o+3)]||0);t.push(u>>16,u>>8&255,255&u)}return t.length-=[0,0,2,1][n],t},c=function(r){for(var t=[],n=0,e=r.length;e>n;n++){var o=r[n];128>o?t.push(o):2048>o?t.push(192|o>>>6,128|63&o):t.push(224|o>>>12&15,128|o>>>6&63,128|63&o)}return t},i=function(r){for(var t=[],n=0,e=r.length;e>n;n++){var o=r[n];if(128>o)t.push(o);else{var a=r[++n];if(224>o)t.push((31&o)<<6|63&a);else{var u=r[++n];t.push((15&o)<<12|(63&a)<<6|63&u)}}}return t},T=function(r){return a(o(r))},f=function(r){return F(u(r))},v=function(r){return i(o(r))},h=function(r){return F(i(r))},s=function(r){return F(i(o(r)))},A=function(r){return c(o(r))},g=function(r){return F(c(r))},U=function(r){return F(c(o(r)))},F=function(r){var t,n=[];for(t=0;t<r.length;t+=65536)n.push(String.fromCharCode.apply(String,r.slice(t,t+65536)));return n.join("")};if(r.btoa)var y=r.btoa,p=function(r){return y(U(r))};else var y=T,p=function(r){return a(A(r))};if(r.atob)var l=r.atob,B=function(r){return s(l(r))};else var l=f,B=function(r){return h(u(r))};r.Base64={convertUTF8ArrayToBase64:a,convertByteArrayToBase64:a,convertBase64ToUTF8Array:u,convertBase64ToByteArray:u,convertUTF16ArrayToUTF8Array:c,convertUTF16ArrayToByteArray:c,convertUTF8ArrayToUTF16Array:i,convertByteArrayToUTF16Array:i,convertUTF8StringToBase64:T,convertBase64ToUTF8String:f,convertUTF8StringToUTF16Array:v,convertUTF8ArrayToUTF16String:h,convertByteArrayToUTF16String:h,convertUTF8StringToUTF16String:s,convertUTF16StringToUTF8Array:A,convertUTF16StringToByteArray:A,convertUTF16ArrayToUTF8String:g,convertUTF16StringToUTF8String:U,convertUTF16StringToBase64:p,convertBase64ToUTF16String:B,fromBase64:f,toBase64:T,atob:l,btoa:y,utob:U,btou:s,encode:p,encodeURI:function(r){return p(r).replace(/[+\/]/g,function(r){return"+"==r?"-":"_"}).replace(/=+$/,"")},decode:function(r){return B(r.replace(/[-_]/g,function(r){return"-"==r?"+":"/"}))}}}(this);


js/zerobin.js
function secondsToHuman(e){if(60>e){var t=Math.floor(e);return t+" second"+(t>1?"s":"")}if(3600>e){var t=Math.floor(e/60);return t+" minute"+(t>1?"s":"")}if(86400>e){var t=Math.floor(e/3600);return t+" hour"+(t>1?"s":"")}if(5184e3>e){var t=Math.floor(e/86400);return t+" day"+(t>1?"s":"")}var t=Math.floor(e/2592e3);return t+" month"+(t>1?"s":"")}function hashToParameterString(e){var t="";for(key in e)""===t?(t=encodeURIComponent(key),t+="="+encodeURIComponent(e[key])):(t+="&"+encodeURIComponent(key),t+="="+encodeURIComponent(e[key]));return t+="&p=p"}function parameterStringToHash(e){for(var t={},n=e.split("&"),s=0;s<n.length;s++){var o=n[s].split("="),i=decodeURIComponent(o[0]),a=decodeURIComponent(o[1]);t[i]=a}return t}function getParameterHash(){var e=window.location.href.indexOf("#");return e>=0?parameterStringToHash(window.location.href.substring(e+1)):{}}function compress(e){return Base64.toBase64(RawDeflate.deflate(Base64.utob(e)))}function decompress(e){return Base64.btou(RawDeflate.inflate(Base64.fromBase64(e)))}function zeroCipher(e,t){return sjcl.encrypt(e,compress(t))}function zeroDecipher(e,t){return decompress(sjcl.decrypt(e,t))}function scriptLocation(){var e=window.location.href.substring(0,window.location.href.length-window.location.search.length-window.location.hash.length),t=e.indexOf("#");return-1!==t&&(e=e.substring(0,t)),e}function pasteID(){return window.location.search.substring(1)}function htmlEntities(e){return String(e).replace(/&/g,"&").replace(/</g,"&lt;").replace(/>/g,"&gt;").replace(/"/g,"&quot;")}function setElementText(e,t){if($("div#oldienotice").is(":visible")){var n=htmlEntities(t).replace(/\n/gi,"\r\n<br>");e.html("<pre>"+n+"</pre>")}else e.text(t)}function applySyntaxColoring(){"<pre><code>"!=$("div#cleartext").html().substring(0,11)&&$("div#cleartext").html("<pre><code>"+$("div#cleartext").html()+"</code></pre>"),hljs.highlightBlock(document.getElementById("cleartext")),$("div#cleartext").css("padding","0")}function displayMessages(e,t){try{var n=zeroDecipher(e,t[0].data)}catch(s){return $("div#cleartext").hide(),$("button#clonebutton").hide(),void showError("Could not decrypt data (Wrong key ?)")}if(setElementText($("div#cleartext"),n),urls2links($("div#cleartext")),t[0].meta.syntaxcoloring&&applySyntaxColoring(),t[0].meta.expire_date&&$("div#remainingtime").removeClass("foryoureyesonly").text("This document will expire in "+secondsToHuman(t[0].meta.remaining_time)+".").show(),t[0].meta.burnafterreading&&($("div#remainingtime").addClass("foryoureyesonly").text("FOR YOUR EYES ONLY.  Don't close this window, this message can't be displayed again.").show(),$("button#clonebutton").hide()),t[0].meta.opendiscussion){$("div#comments").html("");for(var o=1;o<t.length;o++){var i=t[o],n="[Could not decrypt comment ; Wrong key ?]";try{n=zeroDecipher(e,i.data)}catch(s){}var a=$("div#comments"),r="div#comment_"+i.meta.parentid;$(r).length&&(a=$(r));var d=$('<div class="comment" id="comment_'+i.meta.commentid+'"><div class="commentmeta"><span class="nickname"></span><span class="commentdate"></span></div><div class="commentdata"></div><button onclick="open_reply($(this),\''+i.meta.commentid+"');return false;\">Reply</button></div>");setElementText(d.find("div.commentdata"),n),urls2links(d.find("div.commentdata")),d.find("span.nickname").html("<i>(Anonymous)</i>");try{d.find("span.nickname").text(zeroDecipher(e,i.meta.nickname))}catch(s){}d.find("span.commentdate").text("  ("+new Date(1e3*i.meta.postdate).toString()+")").attr("title","CommentID: "+i.meta.commentid),i.meta.vizhash&&d.find("span.nickname").before('<img src="'+i.meta.vizhash+'" class="vizhash" title="Anonymous avatar (Vizhash of the IP address)" />'),a.append(d)}$("div#comments").append('<div class="comment"><button onclick="open_reply($(this),\''+pasteID()+"');return false;\">Add comment</button></div>"),$("div#discussion").show()}}function open_reply(e,t){$("div.reply").remove(),e.after('<div class="reply"><input type="text" id="nickname" title="Optional nickname..." value="Optional nickname..." /><textarea id="replymessage" class="replymessage" cols="80" rows="7"></textarea><br><button id="replybutton" onclick="send_comment(\''+t+'\');return false;">Post comment</button><div id="replystatus">&nbsp;</div></div>'),$("input#nickname").focus(function(){$(this).css("color","#000"),$(this).val()==$(this).attr("title")&&$(this).val("")}),$("textarea#replymessage").focus()}function send_comment(e){if(0!=$("textarea#replymessage").val().length){showStatus("Sending comment...",spin=!0);var t=zeroCipher(pageKey(),$("textarea#replymessage").val()),n="",s=$("input#nickname").val();""!=s&&"Optional nickname..."!=s&&(n=zeroCipher(pageKey(),s));var o={data:t,parentid:e,pasteid:pasteID(),nickname:n};$.post(scriptLocation(),o,"json").error(function(){showError("Comment could not be sent (serveur error or not responding).")}).success(function(e){0==e.status?(showStatus("Comment posted."),location.reload()):showError(1==e.status?"Could not post comment: "+e.message:"Could not post comment.")})}}function send_data(){if(0!=$("textarea#message").val().length){if(!sjcl.random.isReady())return showStatus("Sending paste (Please move your mouse for more entropy)...",spin=!0),void sjcl.random.addEventListener("seeded",function(){send_data()});showStatus("Sending paste...",spin=!0);var e=sjcl.codec.base64.fromBits(sjcl.random.randomWords(8,0),0),t=zeroCipher(e,$("textarea#message").val()),n={data:t,expire:$("select#pasteExpiration").val(),burnafterreading:$("input#burnafterreading").is(":checked")?1:0,opendiscussion:$("input#opendiscussion").is(":checked")?1:0,syntaxcoloring:$("input#syntaxcoloring").is(":checked")?1:0};$.post(scriptLocation(),n,"json").error(function(){showError("Data could not be sent (serveur error or not responding).")}).success(function(t){if(0==t.status){stateExistingPaste();var n=scriptLocation()+"?"+t.id+"#"+e,s=scriptLocation()+"?pasteid="+t.id+"&deletetoken="+t.deletetoken;showStatus(""),$("div#pastelink").html('Your paste is <a id="pasteurl" href="'+n+'">'+n+'</a> <span id="copyhint">(Hit CTRL+C to copy)</span>'),$("div#deletelink").html('<a href="'+s+'">Delete link</a>'),$("div#pasteresult").show(),selectText("pasteurl"),setElementText($("div#cleartext"),$("textarea#message").val()),urls2links($("div#cleartext")),$("input#syntaxcoloring").is(":checked")&&applySyntaxColoring(),showStatus("")}else showError(1==t.status?"Could not create paste: "+t.message:"Could not create paste.")})}}function selectText(e){var t,n,s=document,o=s.getElementById(e);s.body.createTextRange?(t=s.body.createTextRange(),t.moveToElementText(o),t.select()):window.getSelection&&(n=window.getSelection(),t=s.createRange(),t.selectNodeContents(o),n.removeAllRanges(),n.addRange(t))}function stateNewPaste(){$("button#sendbutton").show(),$("button#clonebutton").hide(),$("button#rawtextbutton").hide(),$("div#expiration").show(),$("div#remainingtime").hide(),$("div#burnafterreadingoption").show(),$("div#opendisc").show(),$("div#syntaxcoloringoption").show(),$("button#newbutton").show(),$("div#pasteresult").hide(),$("textarea#message").text(""),$("textarea#message").show(),$("div#cleartext").hide(),$("div#message").focus(),$("div#discussion").hide()}function stateExistingPaste(){$("button#sendbutton").hide(),$("div#oldienotice").is(":visible")?$("button#clonebutton").hide():$("button#clonebutton").show(),$("button#rawtextbutton").show(),$("div#expiration").hide(),$("div#burnafterreadingoption").hide(),$("div#opendisc").hide(),$("div#syntaxcoloringoption").hide(),$("button#newbutton").show(),$("div#pasteresult").hide(),$("textarea#message").hide(),$("div#cleartext").show()}function rawText(){var e=$("div#cleartext").html(),t=document.open("text/html","replace");t.write("<pre>"+e+"</pre>"),t.close()}function clonePaste(){stateNewPaste(),history.replaceState(document.title,document.title,scriptLocation()),showStatus(""),$("textarea#message").text($("div#cleartext").text())}function newPaste(){stateNewPaste(),showStatus(""),$("textarea#message").text("")}function showError(e){$("div#status").addClass("errorMessage").text(e),$("div#replystatus").addClass("errorMessage").text(e)}function showStatus(e,t){if($("div#replystatus").removeClass("errorMessage"),$("div#replystatus").text(e),!e)return void $("div#status").html("&nbsp;");if(""==e)return void $("div#status").html("&nbsp;");if($("div#status").removeClass("errorMessage"),$("div#status").text(e),t){var n='<img src="img/busy.gif" style="width:16px;height:9px;margin:0px 4px 0px 0px;" />';$("div#status").prepend(n),$("div#replystatus").prepend(n)}}function urls2links(e){var t=/((http|https|ftp):\/\/[\w?=&.\/-;#@~%+-]+(?![\w\s?&.\/;#~%"=-]*>))/gi;e.html(e.html().replace(t,'<a href="$1" rel="nofollow">$1</a>'));var t=/((magnet):[\w?=&.\/-;#@~%+-]+)/gi;e.html(e.html().replace(t,'<a href="$1">$1</a>'))}function pageKey(){var e=window.location.hash.substring(1);return i=e.indexOf("="),i>-1&&(e=e.substring(0,i+1)),i=e.indexOf("&"),i>-1&&(e=e.substring(0,i)),"="!==e.charAt(e.length-1)&&(e+="="),e}sjcl.random.startCollectors(),$(function(){if($("input#burnafterreading").change(function(){$(this).is(":checked")?($("div#opendisc").addClass("buttondisabled"),$("input#opendiscussion").attr({checked:!1}),$("input#opendiscussion").attr("disabled",!0)):($("div#opendisc").removeClass("buttondisabled"),$("input#opendiscussion").removeAttr("disabled"))}),$("div#status").text().length>0)return void showStatus($("div#status").text(),!1);if($("div#status").html("&nbsp;"),$("div#cipherdata").text().length>1){if(0==window.location.hash.length)return void showError("Cannot decrypt paste: Decryption key missing in URL (Did you use a redirector or an URL shortener which strips part of the URL ?)");var e=jQuery.parseJSON($("div#cipherdata").text());stateExistingPaste(),displayMessages(pageKey(),e)}else $("div#errormessage").text().length>1?showError($("div#errormessage").text()):newPaste()});


tmp/page.af3906cfde643ae7f290cfdc51cc9342.rtpl.php
<?php if(!class_exists('raintpl')){exit;}?><html>
<head>
<title>0bin.cf</title>
<meta name="robots" content="all" />
<meta name="keywords" content="pastebin,pastebin clone,zerobin"/>
<meta name="description" content="Minimalist and opensource online pastebin where the server has zero knowledge of pasted data. Data is encrypted/decrypted in the browser using 256 bits AES"/>
<link type="text/css" rel="stylesheet" href="css/zerobin.css?<?php echo rawurlencode( $VERSION );?>" /> 
<link href="http://sx.root.sx/favicon.ico" rel="shortcut icon" type="image/x-icon" />
<script src="js/jquery.js?<?php echo rawurlencode( $VERSION );?>"></script>
<script src="js/sjcl.js?<?php echo rawurlencode( $VERSION );?>"></script>
<script src="js/base64.js?<?php echo rawurlencode( $VERSION );?>"></script>
<script src="js/rawdeflate.js?<?php echo rawurlencode( $VERSION );?>"></script>
<script src="js/rawinflate.js?<?php echo rawurlencode( $VERSION );?>"></script>
<script src="js/zerobin.js?<?php echo rawurlencode( $VERSION );?>"></script>
<link type="text/css" rel="stylesheet" href="js/highlight.styles/monokai.css?<?php echo rawurlencode( $VERSION );?>">
<script src="js/highlight.pack.js?<?php echo rawurlencode( $VERSION );?>"></script>

<!--[if lt IE 10]>
<style> body {padding-left:60px;padding-right:60px;} div#ienotice {display:block;} </style>
<![endif]-->

<!--[if lt IE 10]>
<style> div#ienotice {display:block; }  div#oldienotice {display:block; } </style>
<![endif]-->

</head>
  <body>
      <div id="aboutbox">
          Note: This is a test service: 
          Data may be deleted anytime. Kittens will die if you abuse this service.
</div>
    <h1 title="0bin.cf" onclick="window.location.href=scriptLocation();return false;">0bin.cf</h1><br>
    <noscript><div class="nonworking">Javascript is required for 0bin.cf to work.<br>Sorry for the inconvenience.</div></noscript>
    <div id="oldienotice" class="nonworking">0bin.cf requires a modern browser to work.</div>
    <div id="ienotice">Still using Internet Explorer ? &nbsp;Do yourself a favor, switch to a modern browser: 
        <a href="http://www.mozilla.org/firefox/">Firefox</a>, 
        <a href="http://www.opera.com/">Opera</a>, 
        <a href="http://www.google.com/chrome">Chrome</a>, 
        <a href="http://www.apple.com/safari">Safari</a>...
    </div>
    <div id="status"><?php echo $STATUS;?></div>
    <div id="errormessage" style="display:none"><?php echo htmlspecialchars( $ERRORMESSAGE );?></div>
    <div id="toolbar">
    <button id="newbutton" onclick="window.location.href=scriptLocation();return false;" style="display:none;"><img src="img/icon_new.png" width="11" height="15" />New</button>
    <button id="sendbutton" onclick="send_data();return false;" style="display:none;"><img src="img/icon_send.png" width="18" height="15" />Send</button>
    <button id="clonebutton" onclick="clonePaste();return false;" style="display:none;"><img src="img/icon_clone.png" width="15" height="17" />Clone</button>
    <button id="rawtextbutton" onclick="rawText();return false;" style="display:none; "><img src="img/icon_raw.png" width="15" height="15" style="padding:1px 0px 1px 0px;"/>Raw text</button>
      <div id="expiration" style="display:none;">Expires: 
      <select id="pasteExpiration" name="pasteExpiration">
        <option value="5min">5 minutes</option>
        <option value="10min">10 minutes</option>
        <option value="1hour">1 hour</option>
        <option value="1day">1 day</option>
        <option value="1week">1 week</option>
        <option value="1month" selected="selected">1 month</option>
        <option value="1year">1 year</option>
        <option value="never">Never</option>
      </select>
      </div>
      <div id="remainingtime" style="display:none;"></div>
      <div id="burnafterreadingoption" class="button" style="display:none;">
         <input type="checkbox" id="burnafterreading" name="burnafterreading" />
         <label for="burnafterreading">Burn after reading</label>
      </div>
      <div id="opendisc" class="button" style="display:none;">
         <input type="checkbox" id="opendiscussion" name="opendiscussion" />
         <label for="opendiscussion">Open discussion</label>
      </div>
      <div id="syntaxcoloringoption" class="button" style="display:none;">
         <input type="checkbox" id="syntaxcoloring" name="syntaxcoloring" />
         <label for="syntaxcoloring">Syntax coloring</label>
      </div>
    </div>
    <div id="pasteresult" style="display:none;">
      <div id="deletelink"></div>
      <div id="pastelink"></div>
    </div>
    <div id="cleartext" style="display:none;"></div>
    <textarea id="message" name="message" cols="80" rows="25" style="display:none;"></textarea>
    <div id="discussion" style="display:none;">
        <h4>Discussion</h4>
        <div id="comments"></div>
    </div>
    <div id="cipherdata" style="display:none;"><?php echo $CIPHERDATA;?></div>
  </body>

<br><br>
  <center>Powered by <a href="http://www.vaskir.com"><font color="black">Vaskir</font></a> | Made by <a href="https://github.com/sebsauvage/ZeroBin"><font color="black">sebsauvage</font></a><br>Improved version of ZeroBin : <a href="https://github.com/demogorgonz/0bin.cf-PasteBin-clone"><font color="black">DemogorgonZ</font></a> <br><br> <font color="orange">Donate BTC to DemogorgonZ:</font>1FGBdQobW1Cw9u3yapfcro7i7xzYLQKMoF <br> <font color="yellow">Donate DogeCoin to DemogorgonZ:</font> DMus2qFMjrZB1U7ouQ3Eo1RDfnbLjhgEAx </center>

</html>
